criu/util.c

```diff
@@ -1493,4 +1493,90 @@ int cut_path_ending(char *path, char *ending)
 
 	path[ending_pos - 1] = 0;
 	return 0;
+}
+
+static int is_iptables_nft(char *bin)
+{
+	int pfd[2] = { -1, -1 }, ret = -1;
+	char *cmd[] = { bin, "-V", NULL };
+	char buf[100];
+
+	if (pipe(pfd) < 0) {
+		pr_perror("Unable to create pipe");
+		goto err;
+	}
+
+	ret = cr_system(-1, pfd[1], -1, cmd[0], cmd, 0);
+	if (ret) {
+		pr_err("%s -V failed\n", cmd[0]);
+		goto err;
+	}
+
```

Comment:
you need to close pfd[1] here, otherwise, the next call can block forever.

Reply:
thanks, fixed